Sockets/Sockets.Implementation.NET/TcpSocketClient.cs

```diff
@@ -36,11 +38,41 @@ internal TcpSocketClient(TcpClient backingClient)
         /// </summary>
         /// <param name="address">The address of the endpoint to connect to.</param>
         /// <param name="port">The port of the endpoint to connect to.</param>
-        public async Task ConnectAsync(string address, int port)
+        /// <param name="secure">Is this connection secure?</param>
+        public async Task ConnectAsync(string address, int port, bool secure = false)
         {
             await _backingTcpClient.ConnectAsync(address, port);
+            if (!secure)
```

Comment:
Based on what I've read, if you push any changes to your existing master branch, they will automatically update in this PR :)